pkg/querier/querier.go

```diff
@@ -191,6 +209,36 @@ func (q *Querier) SelectSamples(ctx context.Context, params logql.SelectSamplePa
 	return iter.NewMergeSampleIterator(ctx, iters), nil
 }
 
+func (q *Querier) deletesForUser(ctx context.Context, startT, endT time.Time) ([]*logproto.Delete, error) {
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	d, err := q.deleteGetter.GetAllDeleteRequestsForUser(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	start := startT.UnixNano()
+	end := endT.UnixNano()
+
+	var deletes []*logproto.Delete
+	for _, del := range d {
+		for _, selector := range del.Selectors {
+			if int64(del.StartTime) <= end && int64(del.EndTime) >= start {
+				deletes = append(deletes, &logproto.Delete{
```

Comment:
In a future PR it would be nice to have this as a metrics somewhere so we can see if the different instances are in sync.